packaging/opensuse/ksh.spec.in

```diff
@@ -0,0 +1,79 @@
+%global commit #COMMIT#
+%global shortcommit #SHORTCOMMIT#
+%global commitnum #COMMITNUM#
+%global commitdate #COMMITDATE#
+
+Name:           ksh
+Summary:        The Original ATT Korn Shell
+URL:            http://www.kornshell.com/
+License:        EPL
+Version:        %{commitdate}+git.%{commitnum}.%{shortcommit}
+Release:        0%{?dist}
+Source0:        https://github.com/att/ast/archive/%{commit}/%{name}-%{shortcommit}.tar.gz
+
+BuildRequires:  meson
+BuildRequires:  gcc
+BuildRequires:  glibc-devel
+BuildRequires:  ed
+Conflicts:      pdksh
+Requires(post): grep, coreutils
+Requires(postun): sed
+
+Provides:       /bin/ksh
+
+%description
+KSH-93 is the most recent version of the KornShell by David Korn of
+AT&T Bell Laboratories.
+KornShell is a shell programming language, which is upward compatible
+with "sh" (the Bourne Shell).
+
+%prep
+%autosetup -n ast-%{commit}
+
+%build
+%meson -Dbuild-pty=false
+%meson_build
```

Comment:
Out of curiosity what does this line, and `meson_install` do? I don't see any functions in this script with those names and those commands are provided by the Meson project AFAICT.

Reply:
These macros are provided by Meson and are documented [here](https://fedoraproject.org/wiki/Packaging:Meson#Available_Macros).